openedx/core/djangoapps/schedules/management/commands/tests/send_email_base.py

```diff
@@ -92,6 +102,24 @@ def _get_template_overrides(self):
         templates_override[0]['OPTIONS']['string_if_invalid'] = "TEMPLATE WARNING - MISSING VARIABLE [%s]"
         return templates_override
 
+    def _schedule_factory(self, offset=None, **factory_kwargs):
+        _, _, target_day, upgrade_deadline = self._get_dates(offset=offset)
+        factory_kwargs.setdefault('start', target_day)
+        factory_kwargs.setdefault('upgrade_deadline', upgrade_deadline)
+        factory_kwargs.setdefault('enrollment__course__self_paced', True)
+        if hasattr(self, 'experience_type'):
+            factory_kwargs.setdefault('experience__experience_type', self.experience_type)
+        schedule = ScheduleFactory(**factory_kwargs)
+        course_id = schedule.enrollment.course.id
+        if course_id not in self._courses_with_verified_modes:
+            CourseModeFactory(
+                course_id=schedule.enrollment.course.id,
+                mode_slug=CourseMode.VERIFIED,
+                expiration_datetime=datetime.datetime.now(pytz.UTC) + datetime.timedelta(days=30),
+            )
```

Comment:
Do we have a test that makes sure we do _not_ send an mail when the mode is not verified?

Reply:
Just added one. Good catch.